app/assets/javascripts/pretty-text/upload-short-url.js.es6

```diff
@@ -52,6 +51,26 @@ function retrieveCachedUrl($upload, dataAttribute, callback) {
   }
 }
 
+function getAttributeBasedUrl(dataAttribute, cachedUpload) {
+  if (!cachedUpload.url) { return; }
+
+  // non-attachments always use the full URL
+  if (dataAttribute !== "orig-href") {
+    return cachedUpload.url;
+  }
+
+  // attachments should use the full /secure-media-uploads/ URL
+  // in this case for permission checks
+  if (
+    Discourse.SiteSettings.secure_media &&
```

Comment:
Is there a way to avoid using the `Discourse.SiteSettings` global here? It is deprecated and should be passed into the pretty text rendering engine already.

Reply:
Ah yep I had a feeling it might have been deprecated. I was able to get the site settings a different way in a different area; I will change this as well :+1: